packages/gasket-plugin-elastic-apm/lib/middleware.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable spaced-comment */
+// @ts-check
+/// <reference types="../../gasket-plugin-nextjs" />
+
+/**
+ * @typedef {import('@gasket/engine').Gasket} Gasket
+ * @typedef {import('http').IncomingMessage}  Request
+ * @typedef {import('http').ServerResponse}   Response
+ */
+
+const { callbackify } = require('util');
+const apm = require('elastic-apm-node');
+
+/**
+ * Middleware which renames transactions
+ *
+ * @param {Gasket}    gasket  The Gasket engine
+ * @param {Request}   req     The HTTP request being handled
+ * @param {Response}  res     The server response
+ */
+async function transactionRenamingMiddleware(gasket, req, res) {
+  const transaction = apm.currentTransaction;
+  if (!transaction) {
+    return;
+  }
+
+  const requestDetails = { req, res };
+
+  const [name, labels] = await Promise.all([
+    gasket.execWaterfall('transactionName', transaction.name, requestDetails),
+    gasket.execWaterfall('transactionLabels', {}, requestDetails)
+  ]);
```

Comment:
Somewhat related to my naming question, but couldn't these lifecycles just be condensed to a single one, and allow users to directly manage the transaction themselves without any added abstraction?

```suggestion
gasket.execWaterfall('elasticApmTransaction', transaction, requestDetails)
```

That would leave the door open for any other transaction metadata enrichment ([docs](https://www.elastic.co/guide/en/apm/guide/current/data-model-metadata.html)) without requiring additional Gasket lifecycles and supporting code.

Reply:
I like that; we can just pass the transaction object. Will make the change.

Reply:
Merged, @agerard-godaddy